storage/store.go

```diff
@@ -833,6 +835,78 @@ func (s *Store) MergeRange(subsumingRng *Range, updatedEndKey proto.Key, subsume
 	return subsumedRng, nil
 }
 
+func (s *Store) removeAllMetadata(rep *Replica) error {
+	rangeID := rep.Desc().RangeID
+
+	if err := engine.MVCCDelete(s.engine, nil, keys.RangeGCMetadataKey(rangeID), 
+		s.Clock().Now(), nil); err != nil{
+		return util.Errorf("cannot delete gdmeta %s", keys.RangeGCMetadataKey(rangeID))
+	}
+
+	if err := engine.MVCCDelete(s.engine, nil, keys.RangeLastVerificationTimestampKey(rangeID), 
+		s.Clock().Now(), nil); err != nil {
+		return util.Errorf("cannot delete last ts %s", keys.RangeLastVerificationTimestampKey(rangeID))
+	}
+
+	// RaftLogKey 
+	lastIndex, err := rep.LastIndex()
+	if err != nil {
+		return err
+	}
+	// Delete any previously appended log entries which never committed.
+	for i := uint64(0); i <= lastIndex; i++ {
+		if err := engine.MVCCDelete(s.engine, nil, keys.RaftLogKey(rangeID, i), s.Clock().Now(), nil); err != nil {
+			return util.Errorf("cannot delete log %s", keys.RaftLogKey(rangeID, i))
+		}
+	}
+
+	// ResponseCacheKey
+	if err = engine.MVCCDelete(s.engine, nil, keys.ResponseCacheKey(rangeID, nil), s.Clock().Now(), nil); err != nil {
```

Comment:
Rather than enumerating all of these things, I think we could just construct a prefix that matches all `RangeIDKeys`.